src/Analysis/pipeline/parameter_types.py

```diff
@@ -0,0 +1,323 @@
+import abc
+
+class BaseData(metaclass=abc.ABCMeta):
+    """
+    Abbstract class od port types, that define user for
+    validation and translation to string
+    """
+    def __init__(self):
+        self.name = ""
+        """Display name of port"""
+        self.description = ""
+        """Display description of port"""
+    
+    @abc.abstractmethod 
+    def to_string(self, value):
+        """Presentation of type in json yaml"""
+        pass
+
+    def contain(self, data):
+        """
+        return if structure contain set data
+        """
+        return True
+        
+    @abc.abstractmethod 
+    def is_set(self):
+        """
+        return if structure contain real data
+        """
+        pass
+    
+    @abc.abstractmethod 
+    def assigne(self):
+        """
+        Assigne appropriate python variable to data     
+        """
+        pass
+        
+    @abc.abstractmethod 
+    def getter(self):
+        """
+        Return appropriate python variable to data     
+        """
+        pass
+        
+    @property
+    def value(self):
+        return self.getter()
+        
+class Int(BaseData):
+    """
+    Integer
+    """
+    def __init__(self, integer=None):
+        self.integer = integer
+        """value"""
+    
+    def to_string(self, value):
+        """Presentation of type in json yaml"""
+        return str(value)
+      
+    def is_set(self):
+        """
+        return if structure contain real data
+        """
+        return  self.integer is not None
+
+    def assigne(self,  value):
```

Comment:
OK. Takze tohle by melo byt pretizeni '=', asi se bude muset kontrolovat jestli value ma stejny typ. 
Nejde to prirazovani udelat jednou uz v te abstraktni tride? Vsak ta data mohou byt vzdy v promenne data.